content/docs/atom-guide/dev-guide/shaders/azsl/srg-semantics.md

```diff
@@ -1,10 +1,221 @@
-title: Shader Resource Group Semantics
+title: AZSL, Shader Resource Group Semantics (ShaderResourceGroupSemantic).
 description: Learn about AZSL Shader Resource Group Semantics in the Atom Renderer.
-toc: true
-weight: 500
-draft: true
+weight: 100
 ---
 
-{{< placeholder >}}
\ No newline at end of file
+The `ShaderResourceGroupSemantic`, SRG Semantic, for short, defines the order in which descriptor bindings and descriptor sets (descriptor sets are also known as register spaces in HLSL) are emitted for each `ShaderResourceGroup`.  
+  
+    ShaderResourceGroupSemantic <name>
+    {
+        FrequencyId = <0-based index>;
+        [ShaderVariantFallback = <number of bits>;] // (Optional)
+    }
+  
+Conceptually, an SRG Semantic relates to the frequency of change of SRG data. In particular, the render pipeline has a predefined set of SRG Semantics in "&lt;Engine Root&gt;/Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/SrgSemantics.azsli".  
+https://github.com/o3de/o3de/blob/main/Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/SrgSemantics.azsli  
+  
+* `SRG_PerDraw`: Data changes per draw packet. Highest frequency.
+* `SRG_PerObject`: Data changes per object.
+* `SRG_PerMaterial`: Data changes per material.
+* `SRG_PerSubPass`: Data changes per sub-pass.
+* `SRG_PerPass`: Data changes per pass.
+* `SRG_PerPass_WithFallback`: Same as SRG_PerPass, but used for cases where the Pass SRG defines the Fallback key.
+* `SRG_PerView`: Data changes per viewport, because each viewport has its own camera.
+* `SRG_PerScene`: Data changes per scene. Lowest frequency.
+* `SRG_RayTracingGlobal`: Reserved for ray tracing shaders.
+* `SRG_RayTracingLocal_[1..7]`: Reserved for ray tracing shaders.
+  
+## FrequencyId: Defining the order of emission and register space utilization.
+With the value of `FrequencyId` we can define in what order we utilize the register space when compiling SRGs.  
+Consider the following SRG definitions:  
+  
+    ShaderResourceGroupSemantic SRG_PerDraw
+    {
+        FrequencyId = 0;
+    }
+    ShaderResourceGroupSemantic SRG_PerObject
+    {
+        FrequencyId = 1;
+    }
+    
+    ShaderResourceGroup ObjectSrg : SRG_PerObject
+    {
+        int                      m_objectCounter;
+        Texture2D<float4>        m_texture;
+        RWTexture2D<float4>      m_rwtexture;
+        Sampler                  m_sampler;
+    }
+    
+    // Even though "DrawSrg" is declared after "ObjectSrg",
+    // it uses a ShaderResourceGroupSemantic with higher frequency(FrequencyId == 0)
+    // than "ObjectSrg" with lower frequency (FrequencyId == 1).
+    // This means the resources of "DrawSrg" will be emitted before the resources
+    // of "ObjectSrg".
+    ShaderResourceGroup DrawSrg : SRG_PerDraw
+    {
+        int                      m_drawCounter;
+        Texture2D<float4>        m_texture;
+        RWTexture2D<float4>      m_rwtexture;
+        Sampler                  m_sampler;
+    }
```

Comment:
````suggestion
```c
ShaderResourceGroupSemantic SRG_PerDraw
{
    FrequencyId = 0;
}
ShaderResourceGroupSemantic SRG_PerObject
{
    FrequencyId = 1;
}
    
ShaderResourceGroup ObjectSrg : SRG_PerObject
{
    int                      m_objectCounter;
    Texture2D<float4>        m_texture;
    RWTexture2D<float4>      m_rwtexture;
    Sampler                  m_sampler;
}
    
// Even though "DrawSrg" is declared after "ObjectSrg", it uses
// a ShaderResourceGroupSemantic with higher frequency (FrequencyId == 0)
// than "ObjectSrg" with lower frequency (FrequencyId == 1).
// This means the resources of "DrawSrg" will be emitted before the
// resources of "ObjectSrg".
ShaderResourceGroup DrawSrg : SRG_PerDraw
{
    int                      m_drawCounter;
    Texture2D<float4>        m_texture;
    RWTexture2D<float4>      m_rwtexture;
    Sampler                  m_sampler;
}
```
````

Reply:
Fixed